packages/wonder-blocks-popover/components/popover-dialog.js

```diff
@@ -0,0 +1,124 @@
+// @flow
+import * as React from "react";
+import {StyleSheet} from "aphrodite";
+
+import {View} from "@khanacademy/wonder-blocks-core";
+import {TooltipTail} from "@khanacademy/wonder-blocks-tooltip";
+import Color from "@khanacademy/wonder-blocks-color";
+
+import type {
+    Placement,
+    PopperElementProps,
+} from "@khanacademy/wonder-blocks-tooltip";
+
+import PopoverContent from "./popover-content.js";
+import PopoverContentCore from "./popover-content-core.js";
+
+type Props = {|
+    /**
+     * Required to correctly position the elements inside the dialog
+     * @ignore
+     */
+    ...PopperElementProps,
+
+    /**
+     * The content to render inside the dialog.
+     */
+    children:
+        | React.Element<typeof PopoverContent>
+        | React.Element<typeof PopoverContentCore>,
+
+    /**
+     * Called when popper changes its placement
+     */
+    onUpdate: (placement: Placement) => mixed,
+|};
+
+/**
+ * This is an internal component that we use to render the stuff that appears
+ * when a popover shows. It's composed by two elements: The popover content,
+ * that can be of type [PopoverContent](#PopoverContent) or
+ * [PopoverContentCore](#PopoverContentCore), and the
+ * [TooltipTail](#TooltipTail).
+ *
+ * Note that without explicit positioning, the tail will not be centered.
+ */
+export default class PopoverDialog extends React.Component<Props> {
+    componentDidUpdate(prevProps: Props) {
+        if (prevProps.placement !== this.props.placement) {
+            this.props.onUpdate(this.props.placement);
+        }
+    }
+
+    render() {
+        const {
+            placement,
+            children,
+            outOfBoundaries,
+            updateBubbleRef,
+            updateTailRef,
+            tailOffset,
+            style,
+        } = this.props;
+
+        const contentProps = (children.props: any);
+
+        // extract the background color from the popover content
+        const color: $Keys<typeof Color> = contentProps.emphasized
+            ? "blue"
+            : contentProps.color;
+
+        return (
+            <React.Fragment>
+                <View
+                    ref={updateBubbleRef}
+                    data-placement={placement}
+                    style={[
+                        outOfBoundaries && styles.hide,
+                        styles[`content-${placement}`],
+                        style,
+                    ]}
+                >
+                    {children}
+                    <TooltipTail
```

Comment:
I think we should have a single package for Popover and Tooltip, and this internal component should just be called `Tail`.

Reply:
Just an idea.... What do you think about decoupling `Tail` and `TooltipPopper` (`Popper`) to a new private package? e.g. `wonder-blocks-floating`.